actors/builtin/market/deal.go

```diff
@@ -22,7 +26,7 @@ type DealProposal struct {
 	PieceSize       abi.PaddedPieceSize
 	Client          addr.Address
 	Provider        addr.Address
-	ClientSignature acrypto.Signature
+	ClientSignature *acrypto.Signature
```

Comment:
I literally just confirmed with @whyrusleeping that non-pointers is appropriate for all the nested structures. What motivates this change? Why not just use the zero value?